compiler/src/dotty/tools/dotc/transform/ExtensionMethods.scala

```diff
@@ -77,7 +77,7 @@ class ExtensionMethods extends MiniPhase with DenotTransformer with FullParamete
 
           // Create extension methods, except if the class comes from Scala 2
           // because it adds extension methods before pickling.
-          if (!(valueClass.is(Scala2x)))
+          if !valueClass.is(Scala2x, butNot = Scala2Tasty) then
```

Comment:
I fixed this issue in this commit to be able to test the attribute in `scala2-library-tasty-tests/src/Main.scala`